src/app/loadout-builder/processWorker/process.ts

```diff
@@ -517,3 +517,14 @@ function getBaseStatValues(
   // mapping out from stat values to ensure ordering and that values don't fall below 0 from locked mods
   return statValues.map((statHash) => Math.max(baseStats[statHash], 0));
 }
+
+function flattenSets(sets: IntermediateProcessArmorSet[]): ProcessArmorSet[] {
```

Comment:
Just by going off the name should this take in the SetTracker thing, reduce the items to id's and flatten it? At the moment we are looping over it twice right? Basically remove this and do it here (line 299)
```
const finalSets = setTracker.map((set) => set.statMixes.map((mix) => mix.armorSet)).flat();
```